src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -141,6 +141,10 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
             {
                 throw new ArgumentException(SR.Argument_NewMMFAppendModeNotAllowed, nameof(mode));
             }
+            if (mode == FileMode.Truncate)
+            {
+                throw new ArgumentException(SR.Argument_NewMMFTruncateModeNotAllowed, nameof(mode));
```

Comment:
@JeremyKuhne @stephentoub @jkotas  in Ian's absence -- do we care about the breakingness of  allowing `MemoryMappedFile.CreateFromFile(... FileMode.Truncate));` whereas previously (for no obviously good reason) we did not?

It seems to me we should consider this an improvement. If so I can fix that in #17782